internal/runner/command.go

```diff
@@ -0,0 +1,84 @@
+package runner
+
+import (
+	"os"
+	"os/exec"
+	"os/signal"
+	"slices"
+	"syscall"
+
+	"github.com/kballard/go-shellquote"
+)
+
+// runAndForwardSignal runs the command and forwards any signals received to the command.
+func runAndForwardSignal(cmd *exec.Cmd) error {
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+
+	// Create a channel that will be closed when the command finishes.
+	finishCh := make(chan struct{})
+	defer close(finishCh)
+
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	// Start a goroutine to that waits for a signal or the command to finish.
```

Comment:
The channel stuff mostly extracted from the main, the only changes I made is to prevent signal sent by the sub process to the main process to be forwarded back to the sub process. @DrJosh9000 thinks that it is a sensible approach to extract this logic into a separate function.
@DrJosh9000 would you give another quick look, to see if I don't do anything stupid?